### ISSUE TITLE
Sporadic failures observed in CI runs at test_Data . Errors seem to indicate connection failed.

### ISSUE THREAD

**Issue by gsaluja9** [aperture-data/aperturedb-python#257]
1. Example of such run: https://github.com/aperture-data/aperturedb-python/actions/runs/5591061945/jobs/10221621813?pr=249
2. https://github.com/aperture-data/aperturedb-python/actions/runs/5590537696/jobs/10220387469?pr=256

**Comment by drewaogle** [comment]
https://github.com/aperture-data/aperturedb-python/actions/runs/5590537696/jobs/10220387469?pr=256 is a failure I saw with PR 256, rerunning got https://github.com/aperture-data/aperturedb-python/actions/runs/5590537696/jobs/10220770651.

The error at the end is
```
Traceback (most recent call last):
  File "/aperturedata/aperturedb/DaskManager.py", line 51, in process
    db = connnector_type(host=host, port=port,
  File "/aperturedata/aperturedb/Connector.py", line 102, in __init__
    self._connect()
  File "/aperturedata/aperturedb/Connector.py", line 260, in _connect
    raise ConnectionError("Failed to connect to ApertureDB")
ConnectionError: Failed to connect to ApertureDB
2023-07-18 17:32:59,180 - distributed.worker - WARNING - Compute Failed
Key:       ('process-ecc5b4b5ab511713395c8ac2630aae22', 94)
Function:  execute_task
args:      ((subgraph_callable-887d7d91-a318-4a57-bacd-66566df424bc, 'aperturedb', 55555, True, Session(session_token='adbs_n06TOfdYZc03uiqVzP40N7ih21AG9DJY2LU', refresh_token='adbr_BnPJgRv4qmLhgnpmy9PRG1t5bPhp3jJvqDL', session_token_ttl=36000, refresh_token_ttl=360000, session_started=1689701550.8041527), <class 'aperturedb.Connector.Connector'>, 'astype-866ca63ed1c0aeb56cd1209e0e936bc2', 'read-csv-8d42e38dd8d3eee774994ae1c0158d15', [(<function read_block_from_file at 0x7f290819d000>, <OpenFile '/aperturedata/test/input/images.adb.csv'>, 262204, 2789, b'\n'), None, False, False]))
kwargs:    ***
Exception: 'UnboundLocalError("local variable \'db\' referenced before assignment")'
```

and you see some
```
2023-07-18 17:32:59,170 : ERROR : aperturedb.Connector : 139741070616128 : 257 : Error connecting to server: a bytes-like object is required, not 'NoneType'
2023-07-18 17:32:59,171 : ERROR : aperturedb.DaskManager : 139741070616128 : 54 : Failed to connect to ApertureDB
```

also it seems like there are some
```
test/test_Data.py::TestEntityLoader::test_LoaderDescriptorsImages[TCP-with_dask]
  /opt/venv/lib/python3.10/site-packages/distributed/node.py:182: UserWarning: Port 8787 is already in use.
  Perhaps you already have a cluster running?
  Hosting the HTTP server on port 42787 instead
    warnings.warn(
```

which makes me thinks what could be happening is that the next dask run is starting before the old has fully stopped and it might have open connections.

Also saw:

```
test/test_Data.py::TestEntityLoader::test_Loader[HTTP-without_dask]
  /aperturedata/aperturedb/DaskManager.py:34: RuntimeWarning: coroutine 'wait_for' was never awaited
    self._client.close()
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.

test/test_Data.py::TestEntityLoader::test_Loader[HTTP-without_dask]
  /aperturedata/aperturedb/DaskManager.py:35: RuntimeWarning: coroutine 'SpecCluster._close' was never awaited
    self._cluster.close()
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
```

which makes me think the "old dask not fully stopped"

**Comment by gsaluja9** [comment]
[This](https://github.com/aperture-data/aperturedb-python/actions/runs/5594202873/jobs/10228800084?pr=249#step:7:168)  seems to suggest a failure in fixture.
Utils.remove_all_objects was called but it got false.
Looking at remove_all_objects, it is clear that that function executes many queries, which are non atomic and this is legit.

Need to fix utils.
@luisremis

**Comment by luisremis** [comment]
it is ok if it is not a single transaction, but I am not sure why it is returning `False` (failing to delete). looks like it the db is crashing. I don't think we have introduced changes on deletes recently, so not sure why it would start failing now, but it may be the db just crashing on deletes and we will have to fix it.

**Comment by gsaluja9** [comment]
Addressed in #259
